dev.c

```diff
@@ -468,6 +468,16 @@ static ssize_t fuse_dev_do_read(struct fuse_conn *fc, struct file *file,
 		}
 	}
 
+	/* Check if more requests could be picked up */
+	if (remain && request_pending(fc)) {
```

Comment:
Instead  of 2 checks for request_pending() here you can probably just jump to L397 and make the necessary changes under the existing request_pending() check.